packages/dapp-svelte-wallet/react-ui/src/components/tests/WalletConnection.test.js

```diff
@@ -0,0 +1,172 @@
+import { act } from '@testing-library/react';
+import { mount } from 'enzyme';
+import { ApplicationContext } from '../../contexts/Application';
+import WalletConnection from '../WalletConnection';
+
+jest.mock('@agoric/eventual-send', () => ({
+  E: (obj) =>
+    new Proxy(obj, {
+      get(target, propKey) {
+        const method = target[propKey];
+        return (...args) =>
+          new Promise((resolve) => resolve(method.apply(this, args)));
+      },
+    }),
+}));
+
+jest.mock('../../store.js', () => {
+  return { reducer: jest.fn(), setConnectionState: (state) => state };
+});
+
+jest.mock('@agoric/wallet-connection/react.js', () => {
+  return {
+    makeReactAgoricWalletConnection: jest.fn(() => 'wallet-connection'),
+  };
+});
+
+describe('WalletConnection', () => {
+  let component;
+  const dispatch = jest.fn();
+
+  beforeEach(() => {
+    component = mount(
+      <ApplicationContext.Provider value={{ dispatch }}>
+        <WalletConnection />
+      </ApplicationContext.Provider>,
+    );
+  });
+
+  test('dispatches the current connection state', () => {
+    act(() =>
+      component
+        .find('wallet-connection')
+        .props()
+        .onState({ detail: { walletConnection: {}, state: 'connecting' } }),
+    );
+
+    expect(dispatch).toHaveBeenCalledWith('connecting');
+  });
+
+  test('resets the connection on error state', () => {
+    const reset = jest.fn();
+
+    act(() =>
+      component
+        .find('wallet-connection')
+        .props()
+        .onState({ detail: { walletConnection: { reset }, state: 'error' } }),
+    );
+
+    expect(reset).toHaveBeenCalled();
+  });
+
+  describe('on idle state', () => {
+    const accessToken = 'asdf';
+    const getAdminBootstrap = jest.fn();
+    const setItem = jest.fn();
+    const getItem = (_) => `?accessToken=${accessToken}`;
```

Comment:
I had trouble using `jest.fn((_) => `?accessToken=${accessToken})` here. For some reason if I call `getItem` directly it returns the correct value, but 
```
      window.localStorage = {
        setItem,
        getItem,
      };
      `window.localStorage.getItem(...)`
  ```
Properly satisfies `expect(getItemSpy).toHaveBeenCalledWith(...);`, yet it returns `undefined`. Go figure...

Reply:
🤔   Bizarre.